airflow/operators/sftp_to_gcs.py

```diff
@@ -0,0 +1,181 @@
+    :param delegate_to: The account to impersonate, if any
+    :type delegate_to: str
+    :param mime_type: The mime-type string
+    :type mime_type: str
+    :param gzip: Allows for file to be compressed and uploaded as gzip
+    :type gzip: bool
+    :param move_object: When move object is True, the object is moved instead
+        of copied to the new location. This is the equivalent of a mv command
+        as opposed to a cp command.
+    :type move_object: bool
+    """
+
+    template_fields = ("source_path", "destination_path", "destination_bucket")
+
+    @apply_defaults
+    def __init__(
+        self,
+        source_path: str,
+        destination_bucket: str,
+        destination_path: Optional[str] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        sftp_conn_id: str = "ssh_default",
+        delegate_to: Optional[str] = None,
+        mime_type: str = "application/octet-stream",
+        gzip: bool = False,
+        move_object: bool = False,
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.source_path = source_path
+        self.destination_path = self._set_destination_path(destination_path)
+        self.destination_bucket = self._set_bucket_name(destination_bucket)
+        self.gcp_conn_id = gcp_conn_id
+        self.mime_type = mime_type
+        self.delegate_to = delegate_to
+        self.gzip = gzip
+        self.sftp_conn_id = sftp_conn_id
+        self.move_object = move_object
+
+    def execute(self, context):
+        gcs_hook = GoogleCloudStorageHook(
+            gcp_conn_id=self.gcp_conn_id, delegate_to=self.delegate_to
+        )
+
+        sftp_hook = SFTPHook(self.sftp_conn_id)
+
+        if WILDCARD in self.source_path:
+            total_wildcards = self.source_path.count(WILDCARD)
+            if total_wildcards > 1:
+                raise AirflowException(
+                    "Only one wildcard '*' is allowed in source_path parameter. "
+                    "Found {} in {}.".format(total_wildcards, self.source_path)
+                )
+
+            prefix, delimiter = self.source_path.split(WILDCARD, 1)
+            base_path = os.path.dirname(prefix)
+
+            files, _, _ = sftp_hook.get_tree_map(
+                base_path, prefix=prefix, delimiter=delimiter
+            )
+
+            for file in files:
+                destination_path = file.replace(base_path, self.destination_path, 1)
+                self._copy_single_object(gcs_hook, sftp_hook, file, destination_path)
+
+        else:
+            destination_object = os.path.join(self.destination_path, self.source_path)
+            self.log.info("destination_object %s", destination_object)
+            destination_object = (
+                self.destination_path
+                if self.destination_path
+                else self.source_path.rsplit("/", 1)[1]
+            )
+            self.log.info("destination_object %s", destination_object)
```

Comment:
This line looks redundant.

Reply:
Good point. I will fix it and push after verifying by system tests.